backend/src/org/akvo/dash/endpoint/dataset.clj

```diff
@@ -36,27 +36,24 @@
 
 (defn endpoint [{:keys [tenant-manager config]}]
   (context "/api/datasets" {:keys [params tenant] :as request}
+    (let-routes [tenant-conn (connection tenant-manager tenant)]
 
-    (GET "/" []
-      (response (all-datasets (connection tenant-manager tenant))))
+      (GET "/" _
+        (response (all-datasets tenant-conn)))
 
-    (POST "/" {:keys [tenant body jwt-claims] :as request}
-      (let [tenant-conn (connection tenant-manager tenant)]
-        (import/handle-import-request tenant-conn config jwt-claims body)))
+      (POST "/" {:keys [tenant body jwt-claims] :as request}
+        (import/handle-import-request tenant-conn config jwt-claims body))
 
-    (context "/:id" [id]
-
-      (GET "/" []
-        (let [dataset (find-dataset (connection tenant-manager tenant) id)]
-          (if dataset
+      (context "/:id" [id]
+        (GET "/" _
+          (if-let [dataset (find-dataset tenant-conn id)]
             (response dataset)
-            (not-found {:id id}))))
-
-      (DELETE "/" []
-        (delete-dataset-by-id (connection tenant-manager tenant) {:id id})
-        (response {:id id}))
+            (not-found {:id id})))
 
-      (context "/transformations" []
+        (DELETE "/" _
+          (delete-dataset-by-id tenant-conn {:id id})
+          (response {:id id}))
 
-        (GET "/" []
-          (response {:fns []}))))))
+        (context "/transformations" []
```

Comment:
In general, I think we should remove/don't add these kinds of stubs and only add things that we're working on. Before the transformations work is "done" we can't know if we need/want such an endpoint.